src/rabbit_json.erl

```diff
@@ -16,20 +16,53 @@
 
 -module(rabbit_json).
 
--export([decode/1, try_decode/1, encode/1]).
+-export([decode/1, decode/2, try_decode/1, try_decode/2,
+    encode/1, encode/2, try_encode/1, try_encode/2]).
+
 
 -spec decode(jsx:json_text()) -> jsx:json_term().
 decode(JSON) ->
-    jsx:decode(JSON, [return_maps]).
+    decode(JSON, []).
+
+
+-spec decode(jsx:json_text(), jsx_to_term:config()) -> jsx:json_term().
+decode(JSON, Opts) ->
+    jsx:decode(JSON, Opts).
+
 
 -spec try_decode(jsx:json_text()) -> {ok, jsx:json_term()} | error.
 try_decode(JSON) ->
+    try_decode(JSON, []).
+
+
+-spec try_decode(jsx:json_text(), jsx_to_term:config()) -> 
+			{ok, jsx:json_term()} | error.
+try_decode(JSON, Opts) ->
     try
-        {ok, decode(JSON)}
+        {ok, decode(JSON, Opts)}
     catch error:_ ->
-        error
+	    error
```

Comment:
I was wondering if it makes sense to modify the error handling in this way:

```
  catch error: Reason ->
        {error, Reason}
    end.
```
So, in errors like [this](https://github.com/rabbitmq/rabbitmq-server/blob/rabbitmq-common-164/src/rabbit_vhost_limit.erl#L140) we can add more info to understand the problem.